src/common/log.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import log, {LevelOption} from 'electron-log';
+
+import Util from 'common/utils/util';
+
+export const setLoggingLevel = (level: string) => {
+    if (log.transports.file.level === level) {
+        return;
+    }
+    log.error('Logger', 'Log level set to:', level);
+
+    log.transports.console.level = level as LevelOption;
+    log.transports.file.level = level as LevelOption;
+};
+
+export class Logger {
+    private prefixes: string[];
+
+    constructor(...prefixes: string[]) {
+        this.prefixes = prefixes;
+    }
+
+    withPrefix = (...prefixes: string[]) => {
+        return {
+            error: this.prefixed(log.error, ...prefixes),
+            warn: this.prefixed(log.warn, ...prefixes),
+            info: this.prefixed(log.info, ...prefixes),
+            verbose: this.prefixed(log.verbose, ...prefixes),
+            debug: this.prefixed(log.debug, ...prefixes),
+            silly: this.prefixed(log.silly, ...prefixes),
+            log: this.prefixed(log.log, ...prefixes),
+        };
+    }
+
+    private prefixed = (func: (...args: any[]) => void, ...additionalPrefixes: string[]) => {
+        return (...args: any[]) => func(...this.shortenPrefixes(...this.prefixes, ...additionalPrefixes), ...args);
+    }
+
+    private shortenPrefixes = (...prefixes: string[]) => {
+        return prefixes.map((prefix) => `[${Util.shorten(prefix)}]`);
+    }
```

Comment:
Why not do this on creation, so we don't have to repeat it on every log?

If we are going to keep the `withPrefix`, we can just add it to the list.